sklearn/tree/_tree.pyx

```diff
@@ -197,6 +197,16 @@ cdef class Criterion:
 
         pass
 
+    cdef double approx_impurity_improvement(self) nogil:
+        """Approximate the impurity reduction removing constants for a split"""
```

Comment:
+1 to @ogrisel 

`proxy_impurity_improvement` is okay, or `scaled_impurity_improvement` might be better. We don't want to imply it is approximate in any way, since it is not.